bundles/org.openhab.binding.haywardomnilogic/src/main/java/org/openhab/binding/haywardomnilogic/internal/discovery/HaywardDiscoveryService.java

```diff
@@ -0,0 +1,332 @@
+
+                properties.clear();
+                properties.put(HaywardBindingConstants.PROPERTY_TYPE, HaywardTypeToRequest.FILTER);
+                properties.put(HaywardBindingConstants.PROPERTY_SYSTEM_ID, systemIDs.get(i));
+                properties.put(HaywardBindingConstants.PROPERTY_BOWID, bowID.get(i));
+                properties.put(HaywardBindingConstants.PROPERTY_BOWNAME, bowName.get(i));
+                properties.put(HaywardBindingConstants.PROPERTY_FILTER_MINPUMPSPEED, property1.get(i));
+                properties.put(HaywardBindingConstants.PROPERTY_FILTER_MAXPUMPSPEED, property2.get(i));
+                properties.put(HaywardBindingConstants.PROPERTY_FILTER_MINPUMPRPM, property3.get(i));
+                properties.put(HaywardBindingConstants.PROPERTY_FILTER_MAXPUMPRPM, property4.get(i));
+
+                onDeviceDiscovered(HaywardBindingConstants.THING_TYPE_FILTER, names.get(i), properties);
+            }
+
+            // Find Heaters
+            systemIDs = bridgehandler.evaluateXPath("//Heater-Equipment/System-Id/text()", xmlResponse);
+            names = bridgehandler.evaluateXPath("//Heater-Equipment/Name/text()", xmlResponse);
+
+            for (int i = 0; i < systemIDs.size(); i++) {
+                // get Body of Water for each item
+                bowID = bridgehandler.evaluateXPath(
+                        "//*[System-Id=" + systemIDs.get(i) + "]/ancestor::Body-of-water/System-Id/text()",
+                        xmlResponse);
+                bowName = bridgehandler.evaluateXPath(
+                        "//*[System-Id=" + systemIDs.get(i) + "]/ancestor::Body-of-water/Name/text()", xmlResponse);
+
+                properties.clear();
+                properties.put(HaywardBindingConstants.PROPERTY_TYPE, HaywardTypeToRequest.HEATER);
+                properties.put(HaywardBindingConstants.PROPERTY_SYSTEM_ID, systemIDs.get(i));
+                properties.put(HaywardBindingConstants.PROPERTY_BOWID, bowID.get(i));
+                properties.put(HaywardBindingConstants.PROPERTY_BOWNAME, bowName.get(i));
+
+                onDeviceDiscovered(HaywardBindingConstants.THING_TYPE_HEATER, names.get(i), properties);
+            }
+
+            // Find Pumps
+            systemIDs = bridgehandler.evaluateXPath("//Pump/System-Id/text()", xmlResponse);
+            names = bridgehandler.evaluateXPath("//Pump/Name/text()", xmlResponse);
+            names = bridgehandler.evaluateXPath("//Filter/Name/text()", xmlResponse);
+            property1 = bridgehandler.evaluateXPath("//Filter/Min-Pump-Speed/text()", xmlResponse);
+            property2 = bridgehandler.evaluateXPath("//Filter/Max-Pump-Speed/text()", xmlResponse);
+            property3 = bridgehandler.evaluateXPath("//Filter/Min-Pump-RPM/text()", xmlResponse);
+            property4 = bridgehandler.evaluateXPath("//Filter/Max-Pump-RPM/text()", xmlResponse);
```

Comment:
I assume this code was left accidentally.
```suggestion
```

Reply:
Those are actually needed, but weren't put into the properties.  I don't have some of these devices in my backyard so some are untested.